src/stat.cc

```diff
@@ -1684,11 +1684,15 @@ snmpStatGet(int minutes)
     return &CountHist[minutes];
 }
 
-int
-stat5minClientRequests(void)
+bool
+statSawRecentRequests()
 {
-    assert(N_COUNT_HIST > 5);
-    return statCounter.client_http.requests - CountHist[5].client_http.requests;
+    static const int recentMinutes = 5;
+    assert(N_COUNT_HIST > recentMinutes);
+
+    if (NCountHist > recentMinutes)
+        return statCounter.client_http.requests - CountHist[recentMinutes].client_http.requests;
+    return 0;
```

Comment:
The number of recent requests within the first few minutes of Squid process live is not always zero, right?

```suggestion
    return statCounter.client_http.requests;
```

Alternatively, we can use this pattern (that reduces statCounter duplication):

```C++
const auto oldRequests = (NCountHist > recentMinutes) ? CountHist[recentMinutes].client_http.requests : 0;
return statCounter.client_http.requests - oldRequests;
```

Your call.

Reply:
> The number of recent requests within the first few minutes of Squid process live is not always zero

Right, but I thought that since this function cannot produce a meaningful result within these initial few minutes, it should not confuse the caller (which will proceed refreshing DNS). Theoretically, this function could return an 'optional', and the caller would wait in 'zero' and 'undefined' outcomes.

Reply:
Ok.